mllib/src/main/scala/org/apache/spark/ml/tree/treeModels.scala

```diff
@@ -401,8 +403,13 @@ private[ml] object DecisionTreeModelReadWrite {
     }
 
     val dataPath = new Path(path, "data").toString
-    val data = sparkSession.read.parquet(dataPath).as[NodeData]
-    buildTreeFromNodes(data.collect(), impurityType)
+    var df = sparkSession.read.parquet(dataPath)
```

Comment:
I guess spark schema doesn't allow setting default values (and converted from case class) to enable something like below?
spark.read.schema(nodeDataSchema).parquet(dataPath).as[NodeData]

Reply:
CC @zhengruifeng on this question

Reply:
I think it is possible to use `spark.read.schema` here, but need to list all the fields manually (method `schema` needs a `schema: StructType` or `schemaString: String` as input).
So I think it maybe more complicated.

Reply:
you can create the StructType automatically from the case class like:
`Encoders.product[NodeData].schema`
...but I couldn't find a way to utilize default values well, so the new column is all null.

Reply:
@bmarcott Was this change applied for Spark 3.0.1 as well? I am facing the same error in Spark 3.0.1 libs. And I checked the above code inside the imported libraries and could see that the change has not been applied there.

Reply:
It was reported vs 3.0.1, see JIRA. Looks like 3.0.2

Reply:
https://github.com/apache/spark/pull/30889#issuecomment-749872161

see may previous comments, and this fix come in 3.0.2